### ISSUE TITLE
metal attention parity (informational) fails on main: greedy divergence on split-K decode attention path

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#539]
> _Issue filed by Claude (Anthropic agent) on behalf of @ohdearquant._

## Summary

The `metal attention parity (informational)` e2e leg (added in #529) fails on `main` itself, independent of any open PR. 3/4 prompts pass; the fourth diverges from the HF reference at the very first generated token (`agree=0/4`).

Failing run on main (8c73c29): https://github.com/ohdearquant/lattice/actions/runs/28561013518

## Failing case

Prompt (match_window=2):

```
def merge_sort(arr):
    """
    Merge sort implementation.
```

- HF greedy continuation starts: `merge_sort(arr):`
- Lattice greedy continuation starts: `main():`
- `[FAIL] agree=0/4 hf=0.2 lat=0.8 tok/s`

## Path-proof evidence — split-K decode attention is the discriminator

The failing prompt is the **only** one whose path-proof shows the split-K decode attention path active:

| Prompt | prefill_kv_batch | decode_attn_direct | decode_attn_split_partial | decode_attn_split_reduce | Verdict |
|---|---|---|---|---|---|
| `The capital of France is` | 6 | 18 | 0 | 0 | PASS 4/4 |
| `In the year 2024, artificial intelligence` | 6 | 18 | 0 | 0 | PASS 4/4 |
| `def fibonacci(n): ...` | 6 | 18 | 0 | 0 | PASS 4/4 |
| `def merge_sort(arr): ...` | 12 | 0 | 18 | 18 | FAIL 0/4 |

Every prompt that decodes via `decode_attn_direct` matches HF exactly; the one prompt long enough to cross the split-K threshold diverges immediately. This points at the split-K decode attention kernel (partial + reduce) rather than prefill, sampling, or detokenization.

Harness env: `E2E_MAX_TOKENS=4`, `LATTICE_METAL_PATH_PROOF=1`, `LATTICE_BIN=target/release/chat_metal`.

## Suggested reproduction / next steps

1. Reproduce locally with the merge_sort prompt and `LATTICE_METAL_PATH_PROOF=1`; confirm split path is taken.
2. Force `decode_attn_direct` on the same prompt (raise the split threshold) and check whether greedy tokens then match HF — that isolates the split kernel.
3. Differential-test the split-K partial/reduce pair against the direct kernel on identical KV state (same Q, K, V, mask) and compare logits max-diff.

## Notes

- The leg is informational (non-required), so this does not block merges; filing per triage so the signal is not lost.
- Related: #529 (leg introduction), #239.

**Comment by ohdearquant** [comment]
> _Comment authored by Claude (Anthropic agent) on behalf of @ohdearquant._

Duplicate of #535 (accidentally auto-closed by #529's merge, now reopened). The path-proof table from this issue has been carried over there, with a correction: the divergence is at position 0, produced from prefill logits before decode attention runs, so the split-K decode kernels are not the suspect — prefill on long prompts is, per #535's differential triage.
